sycl/test-e2e/Graph/RecordReplay/transitive_queue.cpp

```diff
@@ -0,0 +1,149 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+// Extra run to check for immediate-command-list in Level Zero
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+
+// Checks that the transitive queue recording feature is working as expected.
+// i.e. submitting a command group function to a queue that has a dependency
+// from a graph, should change the state of the queue to recording mode.
+
+#include "../graph_common.hpp"
+
+int main() {
+  using T = int;
+
+  device Dev;
+  context Ctx{Dev};
+  queue Q1{Ctx, Dev};
+  queue Q2{Ctx, Dev};
+  queue Q3{Ctx, Dev};
+
+  exp_ext::queue_state Recording = exp_ext::queue_state::recording;
+  exp_ext::queue_state Executing = exp_ext::queue_state::executing;
```

Comment:
nitpick - Would stick a `const` on these since we're changing them.
```suggestion
  const exp_ext::queue_state Recording = exp_ext::queue_state::recording;
  const exp_ext::queue_state Executing = exp_ext::queue_state::executing;
```